engine/jobmaster/dm/task_manager.go

```diff
@@ -200,22 +187,22 @@ func (tm *TaskManager) GetTaskStatus(taskID string) (runtime.TaskStatus, bool) {
 	return value.(runtime.TaskStatus), true
 }
 
-// check a task runs as expected.
-func taskAsExpected(persistentTask *metadata.Task, taskStatus runtime.TaskStatus) bool {
-	// TODO: when running is expected but task is paused, we may still need return true,
-	// because worker will resume it automatically.
-	// refactor when implement operate task
-	return taskStatus.Stage == metadata.StageFinished || taskStatus.Stage == metadata.StageUnscheduled || persistentTask.Stage == taskStatus.Stage
-}
-
-func (tm *TaskManager) operateTaskMessage(ctx context.Context, taskID string, stage metadata.TaskStage) error {
-	if stage != metadata.StageRunning && stage != metadata.StagePaused {
-		return errors.Errorf("invalid expected stage %d for task %s", stage, taskID)
+func genOp(runtimeStage, expectedStage metadata.TaskStage) dmpkg.OperateType {
+	switch {
+	case runtimeStage == metadata.StageRunning && expectedStage == metadata.StagePaused:
+		return dmpkg.Pause
+	case runtimeStage == metadata.StagePaused && expectedStage == metadata.StageRunning:
```

Comment:
this is duplicated with auto resume in worker?

Reply:
No, auto resume will only resume for metadata.StageError

Reply:
auto resume may need to check if the expected stage is paused, then no need to resume an error task

Reply:
jobmaster: resume task if the task is paused but expected to running.
worker: resume task if the task is error.

> auto resume may need to check if the expected stage is paused

if expected stage is paused, jobmaster will send a pause request, worker will paused it, and set stage to pausing/paused, and then will not auto resume

Reply:
a small improvement is if worker is in error stage, and user now set expected stage to paused, master will not send pause  request immediately. Worker may resume 5 miinutes later and change the stage to running, then master send pause request. We can add this case.